lib/data/datasources/remote/storage_service/firestore_service_impl.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter_master/data/datasources/remote/storage_service/firestore_service.dart';
+
+class FirestoreServiceImpl implements FirestoreService {
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+
+  @override
+  Future<List<Map<String, dynamic>>> getQuestions() async {
+    try {
+      final snapshot = await _firestore.collection('questions').get();
+      return snapshot.docs.map((doc) => doc.data()).toList();
+    } catch (e) {
+      rethrow;
+    }
+  }
+
+  @override
+  Future<void> updateQuestionByText(
+      String questionText, Map<String, dynamic> data) async {
+    try {
+      final snapshot = await _firestore
+          .collection('questions')
+          .where('text', isEqualTo: questionText)
+          .get();
+      for (var doc in snapshot.docs) {
```

Comment:
I don't understand why I have to go through a huge list of documents every time and update them